### PR TITLE
Dsd 24 preparations

### LINKED ISSUE #201: New notebook with step-by-step HydroMT usage
### Version checks

- [x] I have checked that the issue still exists on the latest versions

### Kind of issue

Docs are missing

### Documentation problem

The 1D, 2D and 1D2D notebooks contain duplicate information and miss some basic HydroMT explanation. It can be considered to streamline the notebooks (merge them and include more step-by-step info) or add an extra introductory notebook. This can be used for DSD 2024 as the starting point for the users of the demonstration

### PR DESCRIPTION
**Pull request by shartgring**
## Issue addressed
Fixes #201 

## Explanation
Explain how you addressed the bug/feature request, what choices you made and why.

## Checklist
- [X] Updated tests or added new tests <- no tests needed
- [X] Branch is up to date with `main`
- [X] Tests & pre-commit hooks pass
- [X] Updated documentation if needed
- [X] Updated changelog.rst if needed

## Additional Notes (optional)
For the DSD 2024. The demo will be on Monday the 11th in the afternoon

### CODE REVIEW

**Review by hboisgon — COMMENTED** [review]
Hi @shartgring 
Very nice introduction example! HydroMT core or more plugins could re-use it :)

A few notes left still:

- usually the notebooks are stored on git without results so that when users download them they do not get your results in their notebooks. When building the docs, then the notebooks will be executed and the results will show. Leading to:
- you may want to integrate this notebook in the docs. You have the list of examples: docs/getting_started/example_index.rst
- and usually an extra page for example in how to build a model: docs/user_guide/dflowfm_build.rst

I did not run the notebook myself but I saw your results and there's not so muc code that can go wrong so I assume this is okay!

**Comment by shartgring** [comment]
> Hi @shartgring Very nice introduction example! HydroMT core or more plugins could re-use it :)
> 
> A few notes left still:
> 
> * usually the notebooks are stored on git without results so that when users download them they do not get your results in their notebooks. When building the docs, then the notebooks will be executed and the results will show. Leading to:
> * you may want to integrate this notebook in the docs. You have the list of examples: docs/getting_started/example_index.rst
> * and usually an extra page for example in how to build a model: docs/user_guide/dflowfm_build.rst
> 
> I did not run the notebook myself but I saw your results and there's not so muc code that can go wrong so I assume this is okay!

Good points, thanks! I will also walk with Vera through the code in a couple of minutes so that will also be a good test to see if no errors appear. I have included the new notebook in the docs and also fixed some types when doing it (as part of #198)

**Review by hboisgon — APPROVED** [review]
LGTM!

File: examples/hydromt_tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+   "source": [
+    "!hydromt --models"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "920c45b2",
+   "metadata": {},
+   "source": [
+    "We can use the command `hydromt --help` to get an overview of all the commands that can be accessed throug the command line interface. HydroMT is typically used to `hydromt build` a model using a configuration file. It is also possible to `hydromy update` or `hydromt clip` existing models. The command `hydromt export` is not directly related to model building, but used to exoirt data from a data catalog. The command `hydromt check` is used to validate the configuration file and the linked data catalogs and region settings. This can especially be useful to catch mistakes in the model building process without having to pass all the model building steps, which can take a lot of time for large models.\n",
```
[the remark this answers is not in the document]
Comment (shartgring) [reply examples/hydromt_tutorial.ipynb]: Good ones, should be in the code by now

**Pull request merged**